src/lib/rng/stateful_rng/stateful_rng.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+* (C) 2016 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/stateful_rng.h>
+#include <botan/internal/os_utils.h>
+
+namespace Botan {
+
+void Stateful_RNG::clear()
+   {
+   m_successful_initialization = false;
+   m_bytes_since_reseed = 0;
+   m_last_pid = 0;
+   }
+
+size_t Stateful_RNG::reseed(Entropy_Sources& srcs,
+                            size_t poll_bits,
+                            std::chrono::milliseconds poll_timeout)
+   {
+   size_t bits_collected = RandomNumberGenerator::reseed(srcs, poll_bits, poll_timeout);
+
+   if(bits_collected >= poll_bits)
+      {
+      m_successful_initialization = true;
+      m_bytes_since_reseed = 0;
+      }
+
+   return bits_collected;
+   }
+
+void Stateful_RNG::reseed_from_rng(RandomNumberGenerator& rng, size_t poll_bits)
+   {
+   RandomNumberGenerator::reseed_from_rng(rng, poll_bits);
+
+   m_successful_initialization = true;
+   m_bytes_since_reseed = 0;
+   }
+
+void Stateful_RNG::reseed_check(size_t bytes_requested)
+   {
+   const bool fork_detected = (m_last_pid > 0) && (OS::get_process_id() != m_last_pid);
+
+   m_bytes_since_reseed += bytes_requested;
+   m_last_pid = OS::get_process_id();
+
+   if(!is_seeded() ||
+      fork_detected ||
+      (m_max_output_before_reseed > 0 && m_bytes_since_reseed >= m_max_output_before_reseed))
```

Comment:
@cordney Pretend I don't have all of the NIST SPs memorized and remind us what `reseed_interval = 1` does; just forces a reseed after every single request, regardless of length? I think the same result can be had by setting `m_max_output_before_reseed` to 1, since every (non-empty) request will trigger a reseed.